sys/net/gnrc/network_layer/sixlowpan/frag/rb/gnrc_sixlowpan_frag_rb.c

```diff
@@ -367,30 +372,22 @@ static int _rbuf_add(gnrc_netif_hdr_t *netif_hdr, gnrc_pktsnip_t *pkt,
                                     entry.super,
                                     gnrc_netif_hdr_get_netif(netif_hdr),
                                     &tmp))) {
-                        DEBUG("6lo rbuf minfwd: found route, trying to forward\n");
                         _adapt_hdr(&tmp, page);
-                        int res = _forward_frag(pkt, sizeof(sixlowpan_frag_t),
-                                                vrbe, page);
-
-                        /* prevent intervals from being deleted (they are in the
-                         * VRB now) */
-                        entry.rbuf->super.ints = NULL;
-                        gnrc_pktbuf_release(entry.rbuf->pkt);
-                        gnrc_sixlowpan_frag_rb_remove(entry.rbuf);
-                        return (res == 0) ? RBUF_ADD_SUCCESS : RBUF_ADD_ERROR;
+                        return _forward_uncomp(pkt, rbuf, vrbe, page);
                     }
                 }
             }
         }
         if (IS_USED(MODULE_GNRC_SIXLOWPAN_FRAG_MINFWD)) {
-            DEBUG("6lo rbuf: just do normal reassembly\n");
-            if (gnrc_pktbuf_realloc_data(entry.rbuf->pkt,
-                                         entry.super->datagram_size) != 0) {
-                DEBUG("6lo rbuf minfwd: can't allocate packet data\n");
-                gnrc_pktbuf_release(entry.rbuf->pkt);
-                gnrc_sixlowpan_frag_rb_remove(entry.rbuf);
+            /* all cases to try forwarding with minfwd above failed so just do
+             * normal reassembly. For the `minfwd` case however, we need to
+             * resize `entry.rbuf->pkt`, since we kept the packet allocation
+             * with fragment forwarding as minimal as possible in
+             * `_rbuf_get()` */
+            res = _rbuf_resize_for_reassembly(entry.rbuf);
+            if (res = RBUF_ADD_ERROR) {
                 gnrc_pktbuf_release(pkt);
```

Comment:
`gnrc_pktbuf_release_error()`?

Reply:
Exactly. Even in the reassembly case, this is coming from a different host.